### PR TITLE
Remove event

### PR DESCRIPTION
**Pull request by tvaroglu**
- Changes Implemented:
  - Add `Events#delete` button to user `dashboard` page
  - Add edge case testing for `BackEndFacade` and refactor so the integer values of the array are not hard-coded (were getting issues recording cassettes from localhost vs Heroku)

- Quality Control Checklist:

  - [x] Code adheres to Rubocop styling (if unavoidable infractions, please clarify below)
    - Rubocop still hates our lack of `let` in the feature test hook
  - [x] 100% SimpleCov test coverage (if below, please clarify below)
    - Some `BackEndService` methods that remain stubbed are unttested
  - [x] Last commit has passing Travis build

- Blockers (if applicable):
  - A little unsure of the stub in the dashboard feature test for event deletion... thought process is to stub that method completely, so we don't actually delete records from production DB, when cassettes re-record.

- Next Steps & Additional Notes:

### CODE REVIEW

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class EventsController < ApplicationController
+  def create; end
+
+  def destroy
+    BackEndFacade.delete_event(event_destroy_params)
+    redirect_to dashboard_path(current_user.id)
+    flash[:alert] = 'Workout deleted... but you should schedule a new one ASAP if you want those gains!'
+  end
```
Comment (Scott-Borecki) [inline app/controllers/events_controller.rb:8]: Love the flash alert!

File: spec/features/users/dashboard_spec.rb [hunk 2]
```diff
@@ -225,6 +208,35 @@
...
+            within "#event-#{event.id}" do
+              expect(page).to have_content(event.activity)
+              expect(page).to have_link('Delete')
+            end
+          end
+        end
+      end
+
+      it 'displays a link to delete a workout', :vcr do
+        allow(BackEndService).to receive(:delete_event).and_return(204)
```
Comment (gunnarrunner) [inline spec/features/users/dashboard_spec.rb:230]: Exactly what I was gonna do when I stub out a patch method. So I believe this is fine.

File: spec/facades/back_end_facade_spec.rb [hunk 3]
```diff
@@ -1,25 +1,61 @@
...
+      end
+    end
+
+    context 'when the user does not have any friends' do
+      it 'can return an empty array', :vcr do
+        friends = BackEndFacade.get_user_friends(2)
+
+        expect(friends).to be_empty
+      end
+    end
```
Comment (Scott-Borecki) [inline spec/facades/back_end_facade_spec.rb:20]: Thanks for adding the edge case tests for these!

**Review by gunnarrunner — APPROVED** [review]
Looks good to me! Made a comment on the stubbing out of the service. I also am gonna stub out the service for the patch method. Good work.

**Review by Scott-Borecki — APPROVED** [review]
Looks good to me!  I am not sure how we fully test the `delete` API call in the spec files either.  That would be a good one to ask Bob about.  That might be one we can test in production to confirm it is behaving like we expect.
